WordPress/gutenberg
# Add: Media field changing ui to Dataviews and content preview field to posts and pages

Supersedes: https://github.com/WordPress/gutenberg/pull/65331
Fixes https://github.com/WordPress/gutenberg/issues/63719.
cc: @jameskoster, @jasmussen 

This PR implements the following changes:


- A preview field for posts and pages displaying content similar to patterns and templates.
- Adds the ability to have multiple media fields in the APIs. A field can indicate it is a media field by setting isMediaField to true.
- The user can switch the media field by pressing a new menu available under the preview field.

For some users, most pages don't have a featured image so this PR fixes the issue of having a gray square as a page media as the UI shown most of the time. This users can now switch to show the content preview.

## Testing Instructions
- Go to pages, and select the grid layout.
- Go to the view options, and verify there is "Preview" field shown.
- Verify the preview field has a UI to allow changing its source.
- Change the preview to content and verify it works as expected.

## Screenshot
<img width="1572" alt="Screenshot 2024-11-25 at 17 53 16" src="https://github.com/user-attachments/assets/8a22e9cf-2eb8-40bb-bd68-d49e1c0efa58">
<img width="441" alt="Screenshot 2024-11-25 at 17 52 57" src="https://github.com/user-attachments/assets/f0f6c3e0-3395-45fb-862f-5f599a7d69cc">
